### PR TITLE
[SDK-4141] Add support for Pushed Authorization Requests

### PR DESCRIPTION
**Pull request by evansims**
### Description

This PR adds support for [Pushed Authorization Requests](https://www.rfc-editor.org/rfc/rfc6749) (PAR).

**`Auth0\SDK\Configuration\SdkConfiguration` changes:**
- A new configuration option, `pushAuthorizationRequest` (`bool` type; defaults to `false`) toggles PAR use.

**`Auth0\SDK\API\Authentication\PushedAuthorizationRequest` has been added, with two methods:**
- `post()` builds and sends a PAR request — `POST /oauth/par`.
- `create()` calls `post()` to send a PAR request, and then builds and returns a `/authorize` link for the new PAR request.

**`Auth0\SDK\API\Authentication` changes:**
- `pushedAuthorizationRequest()` convenience method has been added, returning an instance of `Auth0\SDK\API\Authentication\PushedAuthorizationRequest`.

**`Auth0\SDK\Auth0` changes:**
- `login()` has been updated to support PAR.

**`Auth0\SDK\Exception\Authentication\ParResponseException` added:**
- This exception type is raised exclusively when a PAR request fails due to an unexpected response structure.

**Testing and Documentation changes:**
- New unit tests have been added to cover the changes.
- A new example application, `examples/features/authentication/pushed-authorization-request` has been added to demonstrate the feature.

**Example usage:**

```php
<?php

$configuration = new SdkConfiguration(
  pushedAuthorizationRequest: true,
  // ...
);

$sdk = new Auth0($configuration);

$sdk->login();
```

### CODE REVIEW

File: examples/features/authentication/pushed-authorization-request/.env.example [hunk 1]
```diff
@@ -0,0 +1,6 @@
+DOMAIN=
```
Comment (Widcket) [inline examples/features/authentication/pushed-authorization-request/.env.example:1]: Should we add comments explaining what each config is and where to get it/generate it (if applicable)?

File: examples/features/authentication/pushed-authorization-request/index.php [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    $baseUrl = parse_url($currentUrl, PHP_URL_SCHEME);
+    $baseUrl .= '://' . parse_url($currentUrl, PHP_URL_HOST);
+    $baseUrl .= parse_url($currentUrl, PHP_URL_PORT) !== null ? ':' . parse_url($currentUrl, PHP_URL_PORT) : '';
+
+    return $baseUrl;
+}
+
+function getRoute(): array {
+    $currentUrl = 'http://' . $_SERVER['HTTP_HOST'] . $_SERVER['REQUEST_URI'];
+    return array_values(array_filter(explode('/', parse_url($currentUrl, PHP_URL_PATH) ?? '')));
```
Comment (Widcket) [inline examples/features/authentication/pushed-authorization-request/index.php:108]: Doesn't `array_filter` take a callback function?

Comment (evansims) [reply examples/features/authentication/pushed-authorization-request/index.php:108]: Good eye! While `array_filter` optionally takes a callback, a quirk of that function is that if one is not provided, it defaults to filtering out empty or null values from the array, which is what we're looking for here. 😄 Makes things simple!

File: examples/features/authentication/pushed-authorization-request/.env.example (hunk 1 above)
Comment (evansims) [reply examples/features/authentication/pushed-authorization-request/.env.example:11]: Good call out, I've updated it with guidance comments 👍

**Comment by evansims** [comment]
Thanks for the review @Widcket, I've pushed some changes based on your feedback

File: tests/Unit/API/Authentication/PushedAuthorizationRequestTest.php [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+use Auth0\SDK\Exception\Authentication\ParResponseException;
+use Auth0\SDK\Exception\NetworkException;
+use Auth0\Tests\Utilities\HttpResponseGenerator;
+use Auth0\Tests\Utilities\MockDomain;
+use Auth0\Tests\Utilities\TokenGenerator;
+
+uses()->group('authentication', 'authentication.pushedAuthorizationRequest');
+
+beforeEach(function(): void {
+    $this->secret = uniqid();;
```
Comment (Widcket) [inline tests/Unit/API/Authentication/PushedAuthorizationRequestTest.php:18]:
```suggestion
    $this->secret = uniqid();
```
There's a double semicolon here.

**Review by Widcket — APPROVED** [review]

File: tests/Unit/API/Authentication/PushedAuthorizationRequestTest.php (hunk 3 above)
Comment (evansims) [reply tests/Unit/API/Authentication/PushedAuthorizationRequestTest.php]: Fixed! Thanks

**Pull request merged**